openam-core/src/main/java/com/sun/identity/sm/ServiceConfigManagerImpl.java

```diff
@@ -420,7 +420,7 @@ public void objectChanged(String dn, int type) {
         String groupName = "";
         String compName = "";
         if (index > 1) {
-            DN compDn = DN.valueOf(dn.substring(0, index - 1));
+            DN compDn = DN.valueOf(LDAPUtils.newDN(dn).toString().substring(0, index - 1));
```

Comment:
This seems incorrect to me - you are slicing DN based on some index after potentially changing the format of the DN by parsing and stringifying it. Is the DN parsing really necessary?

Reply:
The original value of dn used to be result of `com.sun.identity.common.DNUtils#normalizeDN`.
This method implementation is:
1. create new DN using `LDAPUtils#newDN`
2. call `toString()` on the resulting value
3. call `toLowercase()` on the resulting value
4. return the value
Purpose of this row is to safely produce identical result without the lowercasing.
`index` value is calculated from the `normalizedDn`, but it will work correctly, because value of `dn` and `normalizedDn` are exactly the same except for the casing.

Reply:
I don't like that this is not apparent on the first glance. However the legacy code is very messy so I am OK with this.